lindi/LindiH5ZarrStore/LindiH5ZarrStore.py

```diff
@@ -97,21 +99,19 @@ def from_file(
         if hdf5_file_name_or_url.startswith(
             "http://"
         ) or hdf5_file_name_or_url.startswith("https://"):
+            # note that the remfile.File object does not need to be closed
             remf = remfile.File(hdf5_file_name_or_url, verbose=False)
-            return LindiH5ZarrStore(_file=remf, _url=hdf5_file_name_or_url, _opts=opts)
+            return LindiH5ZarrStore(_file=remf, _url=hdf5_file_name_or_url, _opts=opts, _entities_to_close=[])
         else:
             f = open(hdf5_file_name_or_url, "rb")
-            return LindiH5ZarrStore(_file=f, _url=url, _opts=opts)
+            return LindiH5ZarrStore(_file=f, _url=url, _opts=opts, _entities_to_close=[f])
 
     def close(self):
         """Close the store."""
-        if hasattr(self, "_h5f") and self._h5f:
-            self._h5f.close()
-            self._h5f = None
-        if hasattr(self, "_file") and self._file:
-            if not isinstance(self._file, remfile.File):
-                self._file.close()
-            self._file = None
+        for e in self._entities_to_close:
+            e.close()
+        self._h5f = None
```

Comment:
```suggestion
        self._entities_to_close.clear()
        self._h5f = None
```
Should this reset `self._entities_to_close` as well?

Reply:
Good idea.